ivy_tests/test_ivy/test_frontends/test_tensorflow/test_general_functions.py

```diff
@@ -322,3 +322,73 @@ def test_tensorflow_expand_dims(
         input=value[0],
         axis=axis,
     )
+
+
+# noinspection DuplicatedCode
+@st.composite
+def _arrays_idx_n_dtypes(draw):
+    num_dims = draw(st.shared(helpers.ints(min_value=1, max_value=4), key="num_dims"))
+    num_arrays = draw(
+        st.shared(helpers.ints(min_value=2, max_value=4), key="num_arrays")
+    )
+    common_shape = draw(
+        helpers.lists(
+            arg=helpers.ints(min_value=2, max_value=4),
+            min_size=num_dims - 1,
+            max_size=num_dims - 1,
+        )
+    )
+    unique_idx = draw(helpers.ints(min_value=0, max_value=num_dims - 1))
+    unique_dims = draw(
+        helpers.lists(
+            arg=helpers.ints(min_value=2, max_value=3),
+            min_size=num_arrays,
+            max_size=num_arrays,
+        )
+    )
+    xs = list()
+    input_dtypes = draw(
+        helpers.array_dtypes(
+            available_dtypes=draw(helpers.get_dtypes("numeric")),
+            shared_dtype=True,
+        )
+    )
+    for ud, dt in zip(unique_dims, input_dtypes):
+        x = draw(
+            helpers.array_values(
+                shape=common_shape[:unique_idx] + [ud] + common_shape[unique_idx:],
+                dtype=dt,
+            )
+        )
+        xs.append(x)
+    return xs, input_dtypes, unique_idx
```

Comment:
You may wish to use the helper functions inside the `hypothesis helpers` folder. If I'm not wrong, it is called `dtype_values_axis`.

Reply:
Hi @Aarsh2001, thanks for pointing out that helper function. Please let me know if everything is fine.